pkg/cable/wireguard/driver.go

```diff
@@ -134,26 +148,38 @@ func (w *wireguard) ConnectToEndpoint(remoteEndpoint types.SubmarinerEndpoint) (
 		return "", fmt.Errorf("failed to parse peer public key: %v", err)
 	}
 
-	klog.V(log.DEBUG).Infof("Connecting cluster %s endpoint %s with publicKey %s", remoteEndpoint.Spec.ClusterID, remoteIP, remoteKey)
+	klog.V(log.DEBUG).Infof("Connecting cluster %s endpoint %s with publicKey %s",
+		remoteEndpoint.Spec.ClusterID, remoteIP, remoteKey)
 	w.mutex.Lock()
 	defer w.mutex.Unlock()
 
-	// delete old peers for ClusterID
+	// delete or update old peers for ClusterID
 	oldKey, found := w.peers[remoteEndpoint.Spec.ClusterID]
 	if found {
 		if oldKey.String() == remoteKey.String() {
-			//TODO check that peer config has not changed (eg allowedIPs)
-			klog.V(log.DEBUG).Infof("Skipping update of existing peer key %s", oldKey)
-			return ip, nil
+			// keys match, existing peer, so lets update status
+			if err := w.updatePeerStatus(remoteKey, remoteEndpoint.Spec.ClusterID); err == nil {
+				klog.V(log.DEBUG).Infof("Skipping connect for existing peer key %s", oldKey)
+				return ip, nil
+			}
+			klog.Warningf("Could not update peer %s status, deleting and reconnecting: %v", remoteKey, err)
 		}
 
 		// new peer will take over subnets so can ignore error
 		_ = w.removePeer(&oldKey)
 
 		delete(w.peers, remoteEndpoint.Spec.ClusterID)
+		delete(w.connections, remoteEndpoint.Spec.ClusterID)
 	}
 
+	// create connection, overwrite existing connection
+	connection := v1.NewConnection(remoteEndpoint.Spec)
+	klog.V(log.DEBUG).Infof("Adding connection for cluster %s, %v", remoteEndpoint.Spec.ClusterID, connection)
+	w.connections[remoteEndpoint.Spec.ClusterID] = connection
```

Comment:
We add the `connections` entry here but an intervening error would elide the entry added to the `peers` map below on L212. Maintaining one map as discussed above avoids inconsistency.

Reply:
agree